plugin/storage/es/spanstore/writer.go

```diff
@@ -124,8 +124,9 @@ func getSpanAndServiceIndexFn(p SpanWriterParams) spanAndServiceIndexFn {
 func (s *SpanWriter) WriteSpan(_ context.Context, span *model.Span) error {
 	spanIndexName, serviceIndexName := s.spanServiceIndex(span.StartTime)
 	jsonSpan := s.spanConverter.FromDomainEmbedProcess(span)
+	kind, _ := span.GetSpanKind()
```

Comment:
Why not add Kind field to the dbmodel.Span and not pass it around separately?

Reply:
It will be present as a tag in json model also! I tried fetching kind from original span because of the bool `AllTagsAsFields`. As it will already be present in the json model, should we resave it as a seperate kind also?